### PR TITLE
NMS-14035: Enable key value store access in API plugin

### PR DESCRIPTION
**Pull request by yangli2014**
### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-14035

### CODE REVIEW

File: features/api-layer/common/pom.xml [hunk 1]
```diff
@@ -30,6 +30,7 @@
                 <groupId>org.apache.felix</groupId>
                 <artifactId>maven-bundle-plugin</artifactId>
                 <extensions>true</extensions>
+                <version>5.1.4</version>
```
Comment (cgorantla) [inline features/api-layer/common/pom.xml:33]: version can be specified in parent pom.xml

**Review by cgorantla — COMMENTED** [review]
LGTM, minor comment.

File: features/api-layer/common/src/main/java/org/opennms/features/apilayer/common/distributed/KeyValueStoreWrapper.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+public class KeyValueStoreWrapper<T> implements KeyValueStore<T> {
+    private final org.opennms.features.distributed.kvstore.api.KeyValueStore store;
+
+    public KeyValueStoreWrapper(org.opennms.features.distributed.kvstore.api.KeyValueStore store) {
+        this.store = store;
+    }
+
+    @Override
+    public long put(String key, T value, String context) {
+        return store.put(key, value, context);
```
Comment (j-white) [inline features/api-layer/common/src/main/java/org/opennms/features/apilayer/common/distributed/KeyValueStoreWrapper.java:47]: I think we should add a prefix to the context which would limit what the OIA plugins can touch/access - at worse they could access keys for other plugins.

Comment (yangli2014) [reply features/api-layer/common/src/main/java/org/opennms/features/apilayer/common/distributed/KeyValueStoreWrapper.java:47]: Good idea

File: features/sentinel/repository/pom.xml [hunk 3]
```diff
@@ -600,5 +601,12 @@
             <releases><enabled>false</enabled></releases>
             <snapshots><enabled>true</enabled></snapshots>
         </repository>
+        <repository>
```
Comment (cgorantla) [inline features/sentinel/repository/pom.xml:604]: Do you need this ?

File: .circleci/main/@main.yml [hunk 4]
```diff
@@ -285,7 +285,7 @@ commands:
             .circleci/scripts/configure-signing.sh
             ./clean.pl
             << parameters.node-memory >>
-            export MAVEN_OPTS="$MAVEN_OPTS -Xmx8g -XX:ReservedCodeCacheSize=1g"
+            export MAVEN_OPTS="-Xmx8g -XX:ReservedCodeCacheSize=1g -XX:+TieredCompilation"
```
[the remark this answers is not in the document]
Comment (yangli2014) [reply .circleci/main/@main.yml]: build suceeed

File: features/sentinel/repository/pom.xml (hunk 3 above)
Comment (yangli2014) [reply features/sentinel/repository/pom.xml:604]: Yes for sentinel container build for now.

**Review by j-white — APPROVED** [review]

**Pull request merged**